File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    let numSet = new Set(nums);
    let longest = 0;

    for (const num of numSet) {
        if (!numSet.has(num - 1)) {
            let streak = 1;
            
            while (numSet.has(num + streak)) {
                streak++
            }
            longest = Math.max(longest, streak);
        }
    }
    return longest;
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.go
func longestConsecutive(nums []int) int {
    // make map int -> int
    // loop through build map with num -> bool (exists)
    // a sequence starts if no numSet[key - 1]
    // a sequence continues until there is no numSet[key]
    numSet := make(map[int]bool)
    for _, num := range nums {
        numSet[num] = true
    }

    longest := 0
    for num := range numSet {
        if !numSet[num - 1] {
            streak := 1

            for numSet[num + streak] {
                streak++
            }

            if streak > longest {
                longest = streak
            }
        }
    }
    return longest
}